akka-remote/src/main/scala/akka/remote/artery/driver/UdpTransport.scala

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.remote.artery.driver
+
+import java.net.InetSocketAddress
+
+import akka.NotUsed
+import akka.remote.artery.driver.UdpDriver.Registration
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import akka.stream.scaladsl.Flow
+import akka.stream.stage._
+
+/**
+ * INTERNAL API
+ */
+private[remote] class UdpTransport(driver: UdpDriver) {
+
+  def forRemoteAddress(address: InetSocketAddress): Flow[Frame, Frame, NotUsed] = {
```

Comment:
the part that I don't get with the `Flow` approach is how this would be created and connected for inbound messages, when the system is only receiving messages and not creating outbound associations